src/Microsoft.Health.SqlServer/ManagedIdentityAccessTokenHandler.cs

```diff
@@ -0,0 +1,25 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using Microsoft.Azure.Services.AppAuthentication;
+
+namespace Microsoft.Health.SqlServer
+{
+    public class ManagedIdentityAccessTokenHandler : IAccessTokenHandler
+    {
+        private readonly AzureServiceTokenProvider _azureServiceTokenProvider;
+
+        public ManagedIdentityAccessTokenHandler()
+        {
+            _azureServiceTokenProvider = new AzureServiceTokenProvider();
```

Comment:
We will likely need to be able to pass in the other parameters. One case where this won't work is on things besides the public cloud.
See: https://github.com/Azure/azure-sdk-for-net/blob/6df900655ca8c4afe765ec847e6fab1fdc32193e/sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs#L72

Some examples of possible values this could be:
https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud#azure-ad-authentication-endpoints

Reply:
Updated.